gapic/src/main/com/google/gapid/perfetto/views/BatterySummaryPanel.java

```diff
@@ -168,9 +191,24 @@ public Area getRedraw() {
       public void stop() {
         hovered = null;
       }
+
+      @Override
+      public boolean click() {
+        if ((mods & SWT.MOD1) == SWT.MOD1) {
+          state.addSelection(Kind.Battery, track.getValue(id));
+        } else {
+          state.setSelection(Kind.Battery, track.getValue(id));
+        }
+        return true;
+      }
     };
   }
 
+  @Override
+  public void computeSelection(CombiningBuilder builder, Area area, TimeSpan ts) {
+    builder.add(Selection.Kind.Battery, transform(track.getValues(ts), v -> v));
```

Comment:
What does transform do when the input and output is the same?

Reply:
It's a little hacky way to solve a java generic type issue here I have to admit... Replacing it with only `track.getValues(ts)` will somehow trigger a type unmatch error, because it couldn't sense the class inside the ListenableFuture which indeed implements the correct interface.